### PR TITLE
fix(sdk): queue client should reject empty string message

### PR DESCRIPTION
**Pull request by exoego**
Closes https://github.com/winglang/wing/issues/4142

Queue clients, both for AWS and Simulator, now throw an error if an empty message (`""`) is given.
This is because AWS SQS behaves so.
To ensure the same behavior in implementation for Azure and GCP, the spec is updated too. 

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Review by tsuf239 — COMMENTED** [review]
Awesome! ✨ ✨ 
Can you please test it here too? 
`examples/tests/sdk_tests/queue/push.test.w`
(within a try/catch scope, both when passing a single argument or within a variadic one)

**Comment by exoego** [comment]
@tsuf239 Updated in https://github.com/winglang/wing/pull/4301/commits/039cfe0033c3ae117a8006bf5a62391e5470b625

File: examples/tests/sdk_tests/queue/push.test.w [hunk 1]
```diff
@@ -13,6 +13,14 @@ new std.Test(inflight () => {
     }
   };
 
+  let var error = false;
+  try {
+    q.push("Foo", "");
+  } catch e {
+    error = true;
```
Comment (tsuf239) [inline examples/tests/sdk_tests/queue/push.test.w:20]: Can you please if the error message thrown is "Empty messages are not allowed"?
Can you please add another test case for `q.push("");`? (a single argument)

Comment (exoego) [reply examples/tests/sdk_tests/queue/push.test.w]: Added in https://github.com/winglang/wing/pull/4301/commits/955712415f55a6c9f1501869de82575ea35104d9

**Review by tsuf239 — APPROVED** [review]
Tests have passed, great job!
image

**Pull request merged**